src/data/mod.rs

```diff
@@ -622,3 +622,172 @@ pub fn replace_scalar(a: &mut Array, cond: &Array, b: f64) {
         HANDLE_ERROR(AfError::from(err_val));
     }
 }
+
+/// Create an Array of particular shape filled with given constant.
+///
+/// You can use this function to create arrays of the following types only.
+/// - DType::F32
+/// - DType::C32
+/// - DType::F64
+/// - DType::C64
+/// - DType::B8
+/// - DType::S32
+/// - DType::U32
+/// - DType::U8
+/// - DType::S16
+/// - DType::U16
+///
+/// # Parameters
+///
+/// - `real` is the constant with which output Array is to be filled
+/// - `imag` is the constant with which imaginary part of a complex Array will be filed.
+/// - `dims` is the size of Array
+/// - `dtype` indicates whats the type of the Array to be created
+///
+/// # Return Values
+///
+/// Array of `dims` shape and filed with given constant `value`.
+pub fn create_constant(real: f64, imag: f64, dims: Dim4, dtype: DType) -> Array {
+    unsafe {
+        let mut temp: i64 = 0;
+        let err_val = match dtype {
+            DType::S64 => { AfError::ERR_TYPE as i32 },
+            DType::U64 => { AfError::ERR_TYPE as i32 },
+            DType::C32 => {
+                af_constant_complex(&mut temp as MutAfArray,
+                                    real as c_double, imag as c_double,
+                                    dims.ndims() as c_uint,
+                                    dims.get().as_ptr() as *const DimT, 1)
+            },
+            DType::C64 => {
+                af_constant_complex(&mut temp as MutAfArray,
+                                    real as c_double, imag as c_double,
+                                    dims.ndims() as c_uint,
+                                    dims.get().as_ptr() as *const DimT, 3)
+            },
+            _ => {
+                af_constant(&mut temp as MutAfArray, real as c_double,
+                            dims.ndims() as c_uint,
+                            dims.get().as_ptr() as *const DimT,
+                            dtype as c_int)
+            }
+        };
+        HANDLE_ERROR(AfError::from(err_val));
+        Array::from(temp)
+    }
+}
+
+/// Create an array filled with i64 constant
+///
+/// # Parameters
+///
+/// - `value` is the constant with which output Array is to be filled
+/// - `dims` is the size of Array
+///
+/// # Return Values
+///
+/// Array filled with `value` and `dims` shape.
+pub fn create_constant_i64(value: i64, dims: Dim4) -> Array {
```

Comment:
I see why you made these separate from the `create_constant`  [precision], however can we do something to merge them? I.e. maybe an Enum or something? Not too big a deal either way though.